### ISSUE TITLE
bato.to broken

### ISSUE THREAD

**Issue by Yoposoc** [ameboide/webcomic_reader#48]
bato.to just changed their url patterns.

**Comment by hendrasz** [comment]
Yup, not working right now since they changed the manga's page url.

**Comment by Yoposoc** [comment]
exhentai is also broken.

**Comment by TKW25** [comment]
Don't think it has anything to do with the url, they changed the reader's code entirely.

**Comment by Lunarhawk** [comment]
Use batoto daily, hope someone figures out a fix soon

**Comment by munael** [comment]
Yeah. Can someone knowledgeable try updating bato.to's rules entry in webcomic reader? I feel it could be made to work without changing any plugin-wide code.

**Comment by Tmp341** [comment]
I think batoto changed to user only mode, because manga downloaders started to use batoto username and password.

**Comment by Lunarhawk** [comment]
Yeah you have to be logged in now to view stuff, dunno if that affects it at all tho to my understanding they basically redid the whole reader

**Comment by munael** [comment]
WCR, IIUC, does all its stuff client-side. The switch to user-mode-only shouldn't affect it at all, only the changes to the reader.

**Comment by anka-213** [comment]
It seems like the problem is that bato.to is using ajax to fetch the content, so the image is not there yet when webcomic_reader tries to find it. I'll see if i can find a workaround.

**Comment by hendrasz** [comment]
Manga OnlineViewer's script works on batoto, but reading manga in one page is not really my preference. Maybe you can use their code as example for fetching the page.

**Comment by anka-213** [comment]
That script works in a different way internally*, so I don't think I can use that. 

By the way, what feature(s) from this script do you need/want for bato.to? Their new built in reader seems pretty competent.

<sub>*: It fetches all pages at the same time and grabs the list from the menu, instead of grabbing the link to the next page and going one page at a time.  It also have delays everywhere to compensate for ajax, but that makes stuff slower (obviously).</sub>

**Comment by shogunreaper** [comment]
jdownloader seems to be able to grab all the files, maybe look at their code?

**Comment by munael** [comment]
Right now I miss the minimalist mode the most. Also the ability to quickly page forward _and_ backward by clicking on the image (bato.to ATM only goes forward; although it's now much faster than its older load-on-click behavior).

**Comment by anka-213** [comment]
I have an attempt at a solution now. You can try it at https://github.com/anka-213/webcomic_reader/raw/master/webcomic_reader.user.js.

It is kind of experimental, so please tell me if it breaks something.

**Comment by hendrasz** [comment]
so far it works fine for me

**Comment by Tmp341** [comment]
@anka-213 So, no username and password required for your solution?

**Comment by anka-213** [comment]
@Tmp341 Nope, no login required. I'm not sure what the difference is if you are logged in, but I didn't notice any restrictions.

**Comment by shogunreaper** [comment]
@anka-213 

well you need to be logged in to see the links to the older chapters, only ones from the last 2 weeks or so show up to people not logged in.

though it's not like the links change so if you know them anyone can view it.

**Comment by anka-213** [comment]
@shogunreaper As long as you have a link to one chapter, you can follow the links (or quick chapter navigation) back to the beginning, it seems. But if you don't find any chapter, it seems hard to read anything at all.

**Comment by Lunarhawk** [comment]
Looks to be working fine here, thanks for the fix.

**Comment by tenkuu** [comment]
The script stops working if I navigate to another page or chapter via the navigation bar.
i.e. the `.moderation_bar`

**Comment by anka-213** [comment]
@tenkuu Yes. Or more specifically, it doesn't actually navigate to the new page, it just updates the url. This is a known issue. A workaround is to press enter in the url bar after using the navigation bar.

**Comment by munael** [comment]
Latest script version, running on latest Chrome version (as of this post) (without Tampermonkey), is completely broken with bato.to.

Minimalistic mode doesn't work unless I open the settings window and 'save' again with 'minimalistic' option set. This makes it look like the mode is turned on but in fact you can't advance pages in either direction (both ways the counter is '0...').

Sometimes the url just changes to 'bato.to/reader' and nothing shows.

In normal mode none of the controls actually do anything. The configuration buttons for webcomic-reader show at the bottom of the page but only "Webcomic Reader - Settings" and "Webcomic Reader - Disable for this site" show up. No bookmarks button or a chap/page selector at the bottom, etc.

**Comment by anka-213** [comment]
@Enamex Just to be clear, amiboide's latest version or mine?

**Comment by munael** [comment]
@anka-213 Well, that was confusing haha. I only tried the repo's posted script (updated '10 months ago') so amiboide's, I guess.

Trying yours now (" https://github.com/anka-213/webcomic_reader/raw/master/webcomic_reader.user.js ").
